plugins/scaffolder-node/src/checkpoints/types.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { JsonValue } from '@backstage/types';
+
+/**
+ * The status of a checkpoint, indicating whether it succeeded or failed.
+ *
+ * @public
```

Comment:
I think that these should probably be `@alpha` types exported from a `/alpha` endpoint for now.

Don't really want to add these to the public API until we're ready :pray: